### ISSUE TITLE
[tool_crash] XmlNodeTypeException

### ISSUE THREAD

**Issue by Hoai-Phong** [flutter/flutter#108535]
## Command
```
flutter pub get
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
XmlNodeTypeException
```
#0      XmlNodeTypeException.checkValidType (package:xml/src/xml/exceptions/type_exception.dart:14:7)
#1      XmlNodeList.add (package:xml/src/xml/utils/node_list.dart:47:28)
#2      _XmlNodeDecoderSink.commit (package:xml/src/xml_events/converters/node_decoder.dart:147:24)
#3      _XmlNodeDecoderSink.visitDeclarationEvent (package:xml/src/xml_events/converters/node_decoder.dart:78:7)
#4      XmlDeclarationEvent.accept (package:xml/src/xml_events/events/declaration.dart:18:51)
#5      XmlEventVisitor.visit (package:xml/src/xml_events/visitor.dart:14:39)
#6      Iterable.forEach (dart:core/iterable.dart:325:35)
#7      XmlNodeDecoder.convertIterable (package:xml/src/xml_events/converters/node_decoder.dart:53:12)
#8      new XmlDocument.parse (package:xml/src/xml/nodes/document.dart:34:41)
#9      AndroidProject.computeEmbeddingVersion (package:flutter_tools/src/project.dart:649:30)
#10     AndroidProject.getEmbeddingVersion (package:flutter_tools/src/project.dart:628:12)
#11     PackagesGetCommand.usageValues (package:flutter_tools/src/commands/packages.dart:111:67)
#12     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1371:21)
<asynchronous suspension>
#13     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1209:27)
<asynchronous suspension>
#14     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#15     CommandRunner.runCommand (package:args/command_runner.dart:209:13)
<asynchronous suspension>
#16     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
<asynchronous suspension>
#17     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#18     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
```
```
[✓] Flutter (Channel beta, 3.3.0-0.1.pre, on Microsoft Windows [Version 10.0.22622.290], locale en-US)
    • Flutter version 3.3.0-0.1.pre on channel beta at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d952ca86d6 (19 hours ago), 2022-07-27 10:33:00 -0700
    • Engine revision fd131c385e
    • Dart version 2.18.0 (build 2.18.0-271.4.beta)
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\ADMIN\AppData\Local\Android\Sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = C:\Users\ADMIN\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[!] Visual Studio - develop for Windows (Visual Studio Professional 2022 17.1.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Professional
    • Visual Studio Professional 2022 version 17.1.32421.90
    ✗ Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the "Desktop development with C++" workload, and include these components:
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions available, install the latest
        C++ CMake tools for Windows
        Windows 10 SDK

[✓] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.69.2)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.34.0

[✓] Connected device (4 available)
    • M2101K6G (mobile) • d2857e3e • android-arm64  • Android 12 (API 31)
    • Windows (desktop) • windows  • windows-x64    • Microsoft Windows [Version 10.0.22622.290]
    • Chrome (web)      • chrome   • web-javascript • Google Chrome 103.0.5060.134
    • Edge (web)        • edge     • web-javascript • Microsoft Edge 103.0.1264.71

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```

## Flutter Application Metadata
**Type**: app
**Version**: 1.0.0+1
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: stable
**Creation framework version**: 18116933e77adc82f80866c928266a5b4f1ed645

**Comment by danagbemava-nc** [comment]
Hi @Hoai-Phong, does this crash repro with every project you create, or is it just with this project?

Can you also share your `pubspec.yaml`?

Thank you

**Comment by Hoai-Phong** [comment]
hi @danagbemava-nc  , Sorry then i re-download version 3.0.5 problem solved.
description: A new Flutter project.

# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.16.1 <3.0.0"

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2
  http: ^0.13.4
  provider: ^6.0.1
  socket_io_client: ^1.0.2
  hexcolor: ^2.0.5
  web_socket_channel: ^2.1.0
  fluttertoast: ^8.0.8
  device_info_plus: ^3.1.1
  flutter_local_notifications: ^9.1.4
  # timezone: ^0.8.0
  # flutter_background_service: ^0.1.5
  shared_preferences: ^2.0.9
  url_launcher: ^6.0.20
dev_dependencies:
  flutter_test:
    sdk: flutter

  # The "flutter_lints" package below contains a set of recommended lints to
  # encourage good coding practices. The lint set provided by the package is
  # activated in the `analysis_options.yaml` file located at the root of your
  # package. See that file for information about deactivating specific lint
  # rules and activating additional ones.
  flutter_lints: ^1.0.0

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  assets: 
   - assets/images/

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  fonts:
    - family: OpenSans
      fonts:
        - asset: assets/fonts/OpenSans.ttf

    - family: NotoSansJPBlack
      fonts:
        - asset: assets/fonts/NotoSansJPBlack.otf

    - family: NotoSansJPBold
      fonts:
        - asset: assets/fonts/NotoSansJPBold.otf

    - family: NotoSansJPLight
      fonts:
        - asset: assets/fonts/NotoSansJPLight.otf

    - family: NotoSansJPMedium
      fonts:
        - asset: assets/fonts/NotoSansJPMedium.otf

    - family: NotoSansJPRegular
      fonts:
        - asset: assets/fonts/NotoSansJPRegular.otf

    - family: NotoSansJPThin
      fonts:
        - asset: assets/fonts/NotoSansJPThin.otf

  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages

![image](https://user-images.githubusercontent.com/83800555/181560337-bf4dfefd-336d-4792-a77a-a3d9f6f62865.png)

**Comment by danagbemava-nc** [comment]
Thanks for info. 

Labeling so that the team is aware of this issue.

**Comment by christopherfujino** [comment]
This is the same as https://github.com/flutter/flutter/issues/106365 (even though the exception is different, the fix was the same) and was fixed in https://github.com/flutter/flutter/pull/107574, which unfortunately landed after the 3.3.0 beta.

**Comment by christopherfujino** [comment]
@Hoai-Phong However, the "fix" was just giving a more descriptive error message, the underlying cause is that you have a XML typo in your app/src/main/AndroidManifest.xml file. You will need to fix this yourself.
